src/Congo.Contracts/Clients/ICongoUserClient.cs

```diff
@@ -14,5 +15,11 @@ public interface ICongoUserClient
 
         [Post("/api/products/{id}/purchase")]
         Task<ApiResponse<OrderConfirmationResponse>> PurchaseProduct(Guid id);
+
+        [Get("/api/Cart/{id}")]
+        Task<ApiResponse<CartResponse>> GetCart(Guid id);
+
+        [Post("/add-to-cart")]
```

Comment:
Any reason for this not to be `/api/cart/{cartId}` for the GET and POST? The URL format is generally meant to reflect what resource you're operating on (in this case, the specific cart)

Reply:
Currently, "add-to-cart" also creates the cart if empty. If you don't have a cart, you don't have a cart ID.

Nothing is set in stone, so if you can think of a nice way to handle creation of carts, feel free to suggest it :)

Reply:
Ah, I see. Thank you for explaining!
One pattern that I've seen before is to have the "update" endpoint allow for an empty identifier which would trigger a creation. This way, the route still ends up looking similar: `api/cart` as the create and `api/cart/{id}` being the update. The request/response payloads are the same but the former would validate that there is no `id` field set on the request object.
You definitely don't have to listen to this feedback, just figured I'd add some of what I've seen

Reply:
What you've seen before is what we call "RESTful" API design. I agree that what we have right now is.. not that.

A more restful approach would be something like this:

`POST api/cart` - push a product payload here to create a new cart.
`POST api/cart/{id}` - push a product payload here to add it to that cart id.

Reply:
Lol yeah, I'm actually very familiar with REST. Regarding the pattern, I'm pretty sure our proposals are the same. 
The reason I brought it up was because REST is pretty industry standard so definitely trying to follow it would be a beneficial learning experience.
If you're only focusing on this from a web based approach, another option is to use a cookie to track the ID of the cart and keep referencing that instead. This can become more complicated when a user signs in and wants to access their cart on other machines but I'm pretty sure thats well beyond the scope of this exercise.

Reply:
Shall I leave this as it is for now then? If we decide to change the API endpoint at a later date, it will be easy enough to update

Reply:
For sure, don't consider my comments as blockers as I'm mostly a spectator on this project 😃 If I feel strongly about something, I'll be sure to make it known but I'd default all final decisions to the rest of the collaborators. My comments are either to provide additional perspective or for me to gain a better understanding of the intended direction 
- zdev9891